### ISSUE TITLE
Nessus file ingestion

### ISSUE THREAD

**Issue by Xzeryn** [Xzeryn/Elastic-Stack-Docker#15]

**Comment by allamiro** [comment]
These projects (https://github.com/nicpenning/Power-Nessie and https://github.com/iwikmai/Nessus-ES) provide a good starting point for working with Nessus scan data and ElasticSearch.

Instead of using PowerShell, one can leverage bash scripts along with xmlstarlet to  parse the Nessus XML files. The parsed data can then be transformed into JSON using jq  making it compatible for ingestion into ElasticSearch through the _bulk API. This approach offers a streamlined cross-platform solution without relying on PowerShell.

**Comment by nicpenning** [comment]
Sounds doable. Remember PowerShell 7 is cross platform and much easier to install and use then containers as a whole (in my humble opinion) 😀 but would love to see what you can do here. I can see where bash would be supported in more places, though.

The bigger challenge is the APIs to download the files. I was thinking someday using the CEL input and making this an integration to do all of this work for parsing but that is very complicated where as PowerShell is very straightforward and much easier to read and digest for those who are not primarily developers.

**Comment by allamiro** [comment]
> Sounds doable. Remember PowerShell 7 is cross platform and much easier to install and use then containers as a whole (in my humble opinion) 😀 but would love to see what you can do here. I can see where bash would be supported in more places, though.
> 
> The bigger challenge is the APIs to download the files. I was thinking someday using the CEL input and making this an integration to do all of this work for parsing but that is very complicated where as PowerShell is very straightforward and much easier to read and digest for those who are not primarily developers.

Yes, you’re right, that makes a lot of sense! PowerShell 7 being cross-platform and easier to use certainly has its advantages, especially for straightforward implementation and readability. While I see the value in using bash and XML tools for broader support, I agree that PowerShell’s simplicity and accessibility for non-developers are compelling factors. I guess my bias toward sticking with Linux sometimes outweighs the need for simplicity..
